src/openzeppelin/token/erc721/erc721.cairo

```diff
@@ -0,0 +1,462 @@
+    fn transferFrom(from: ContractAddress, to: ContractAddress, tokenId: u256) {
+        ERC721CamelImpl::transferFrom(from, to, tokenId)
+    }
+
+    #[external]
+    fn safe_transfer_from(
+        from: ContractAddress, to: ContractAddress, token_id: u256, data: Span<felt252>
+    ) {
+        ERC721Impl::safe_transfer_from(from, to, token_id, data)
+    }
+
+    #[external]
+    fn safeTransferFrom(
+        from: ContractAddress, to: ContractAddress, tokenId: u256, data: Span<felt252>
+    ) {
+        ERC721CamelImpl::safeTransferFrom(from, to, tokenId, data)
+    }
+
+    // Internal
+
+    #[internal]
+    fn initializer(name_: felt252, symbol_: felt252) {
+        _name::write(name_);
+        _symbol::write(symbol_);
+        erc165::ERC165::register_interface(erc721::interface::IERC721_ID);
+        erc165::ERC165::register_interface(erc721::interface::IERC721_METADATA_ID);
+    }
+
+    #[internal]
+    fn _owner_of(token_id: u256) -> ContractAddress {
+        let owner = _owners::read(token_id);
+        match owner.is_zero() {
+            bool::False(()) => owner,
+            bool::True(()) => panic_with_felt252('ERC721: invalid token ID')
+        }
+    }
+
+    #[internal]
+    fn _exists(token_id: u256) -> bool {
+        !_owners::read(token_id).is_zero()
+    }
+
+    #[internal]
+    fn _is_approved_or_owner(spender: ContractAddress, token_id: u256) -> bool {
+        let owner = _owner_of(token_id);
+        owner == spender | is_approved_for_all(owner, spender) | spender == get_approved(token_id)
+    }
+
+    #[internal]
+    fn _approve(to: ContractAddress, token_id: u256) {
+        let owner = _owner_of(token_id);
+        assert(owner != to, 'ERC721: approval to owner');
+        _token_approvals::write(token_id, to);
+        Approval(owner, to, token_id);
+    }
+
+    #[internal]
+    fn _set_approval_for_all(owner: ContractAddress, operator: ContractAddress, approved: bool) {
+        assert(owner != operator, 'ERC721: self approval');
+        _operator_approvals::write((owner, operator), approved);
+        ApprovalForAll(owner, operator, approved);
+    }
+
+    #[internal]
+    fn _mint(to: ContractAddress, token_id: u256) {
+        assert(!to.is_zero(), 'ERC721: invalid receiver');
+        assert(!_exists(token_id), 'ERC721: token already minted');
+
+        // Update balances
+        _balances::write(to, _balances::read(to) + 1.into());
+
+        // Update token_id owner
+        _owners::write(token_id, to);
+
+        // Emit event
+        Transfer(Zeroable::zero(), to, token_id);
+    }
+
+    #[internal]
+    fn _transfer(from: ContractAddress, to: ContractAddress, token_id: u256) {
+        assert(!to.is_zero(), 'ERC721: invalid receiver');
+        let owner = _owner_of(token_id);
+        assert(from == owner, 'ERC721: wrong sender');
+
+        // Implicit clear approvals, no need to emit an event
+        _token_approvals::write(token_id, Zeroable::zero());
+
+        // Update balances
```

Comment:
Our policy regarding comments is to add them when they clarify stuff that can be confusing, while avoiding redundancy right? Should we remove the obvious ones like this one or "Emit event"?

Reply:
agree on the event one. the balances' one I take it as a grouping comment, meaning it's explaining how the lines are grouped with breaklines

Reply:
We have the same balances comment on line 409 with just one line.